### PR TITLE
Replace keytar with SecretStorage

### LINKED ISSUE #1045: Fail to detach a storage account with an error
**OS:** Windows / Linux
**Azure Resource Extension Build Version:** 20220505.12 (unified)
**Azure Storage Extension Build Version:** 20220505.2 (unified)
**Regression:** Not a regression

**Repro Steps:**
1. Attached a storage account -> Right click this storage account -> Click "Detach Storage Account".
2. Check whether succeeds to detach this storage account.

**Expect:**
Succeed to detach this storage account.

**Actual:**
Fail to detach this storage account with an error.
image

**More Info:**

1. There is an error under the "Attached Storage Accounts" node after refreshing the workspace view.
image
2. This issue doesn't reproduce on Mac.

### PR DESCRIPTION
**Pull request by alexweininger**
Fixes the Attached Storage Accounts tree item
Fixes #1045

### CODE REVIEW

File: src/tree/AttachedStorageAccountsTreeItem.ts [hunk 1]
```diff
@@ -27,13 +26,11 @@ export class AttachedStorageAccountsTreeItem extends AzExtParentTreeItem {
     private _root: ISubscriptionContext;
     private _attachedAccounts: AttachedStorageAccountTreeItem[] | undefined;
     private _loadPersistedAccountsTask: Promise<AttachedStorageAccountTreeItem[]>;
-    private _keytar: KeyTar | undefined;
-    private readonly _serviceName: string = "ms-azuretools.vscode-azurestorage.connectionStrings";
+    private readonly _serviceName: string = "ms-azuretools.vscode-azurestorage.connectionStrings2";
```
Comment (alexweininger) [inline src/tree/AttachedStorageAccountsTreeItem.ts:29]: The storage accounts in the existing globalState will not work. Easiest way to clear them out is to just change the key we're using for globalState.

File: package.json [hunk 2]
```diff
@@ -762,7 +762,6 @@
         "@microsoft/vscode-azext-dev": "^0.1.4",
         "@types/fs-extra": "^8.1.0",
         "@types/gulp": "^4.0.6",
-        "@types/keytar": "4.0.1",
         "@types/mocha": "^7.0.2",
         "@types/node": "^14.0.0",
         "@types/p-retry": "^3.0.1",
```
Comment (bwateratmsft) [inline package.json:767]: I noticed that `@types/mime` is in the `dependencies` section. It's unrelated to this PR, but can you move it up to `devDependencies` and do `npm install` again to update `package-lock.json`?

File: src/tree/AttachedStorageAccountsTreeItem.ts [hunk 3]
```diff
@@ -81,10 +78,8 @@ export class AttachedStorageAccountsTreeItem extends AzExtParentTreeItem {
         const index = attachedAccounts.findIndex((account) => account.fullId === treeItem.fullId);
         if (index !== -1) {
             attachedAccounts.splice(index, 1);
-            if (this._keytar) {
-                await this._keytar.deletePassword(this._serviceName, treeItem.fullId);
-                await this.persistIds(attachedAccounts);
-            }
+            await ext.context.secrets.delete(this._serviceName + treeItem.fullId);
```
Comment (bwateratmsft) [inline src/tree/AttachedStorageAccountsTreeItem.ts:81]: Service name and tree ID are concatenated like this in three places--just to make sure these never get out of sync, I'd create a method to do that.

File: src/tree/AttachedStorageAccountsTreeItem.ts (hunk 1 above)
Comment (bwateratmsft) [reply src/tree/AttachedStorageAccountsTreeItem.ts:29]: Will there be impacts on user experience?

Comment (alexweininger) [reply src/tree/AttachedStorageAccountsTreeItem.ts:29]: No impact on user experience.

**Review by bwateratmsft — APPROVED** [review]

**Pull request merged**